sqlglot/parser.py

```diff
@@ -6841,7 +6841,7 @@ def _parse_merge(self) -> exp.Merge:
             using=using,
             on=on,
             expressions=self._parse_when_matched(),
-            returning=self._match(TokenType.RETURNING) and self._parse_csv(self._parse_bitwise),
+            returning=self._parse_returning()        
```

Comment:
Is there any reason not to be using this here?

Reply:
Probably an oversight? Given no tests failed this is probably fine.